src/integration/integration_api_broker_graceful_shutdown_test.go

```diff
@@ -0,0 +1,177 @@
+package integration_test
+
+import (
+	"fmt"
+	. "integration"
+
+	"net"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/ghttp"
+)
+
+var _ = Describe("Integration_Api_Broker_Graceful_Shutdown", func() {
+
+	const (
+		MessageForServer = "message_for_server"
+	)
+
+	BeforeEach(func() {
+		fakeScheduler = ghttp.NewServer()
+		fakeApiServer := ghttp.NewServer()
+
+		apiServerConfPath = components.PrepareApiServerConfig(components.Ports[APIServer], dbUrl, fakeScheduler.URL(), tmpDir)
+		serviceBrokerConfPath = components.PrepareServiceBrokerConfig(components.Ports[ServiceBroker], brokerUserName, brokerPassword, dbUrl, fakeApiServer.URL(), brokerApiHttpRequestTimeout, tmpDir)
+
+	})
+
+	AfterEach(func() {
+		stopAll()
+	})
+
+	Describe("Graceful Shutdown", func() {
+
+		Context("ApiServer", func() {
+
+			BeforeEach(func() {
+				startApiServer()
+			})
+
+			It("stops receiving new connections after being interrupted", func() {
```

Comment:
Are you suggesting that we also check that the apiserver has gone down; as of now we're only checking that it's not able to receive any new connections ?